src/Controller/RdfMappingsReportController.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace Drupal\islandora\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\taxonomy\Entity\Vocabulary;
+use Drupal\taxonomy\Entity\Term;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+
+/**
+ * RDF mappings report controller.
+ */
+class RdfMappingsReportController extends ControllerBase {
+
+  /**
+   * Output the RDF mappings report.
+   *
+   * @return string
+   *   Markup of the tables.
+   */
+  public function main() {
+    $markup = '';
+
+    // Configured namespaces.
+    $namespaces = rdf_get_namespaces();
+    $namespaces_table_rows = [];
+    foreach ($namespaces as $alias => $namespace_uri) {
+      $namespaces_table_rows[] = [$alias, $namespace_uri];
+    }
+    $namespaces_table_header = [t('Namespace alias'), t('Namespace URI')];
+    $namespaces_table = [
+      '#theme' => 'table',
+      '#header' => $namespaces_table_header,
+      '#rows' => $namespaces_table_rows,
+    ];
+    $namespaces_table_markup = \Drupal::service('renderer')->render($namespaces_table);
+
+    $markup .= '<details><summary>' . t('RDF namespaces used in field mappings') .
+      '</summary><div class="details-wrapper">' . $namespaces_table_markup . '</div></details>';
+
+    // Node and media field to RDF property mappings.
+    $entity_types = ['node', 'media'];
+    $markup .= '<h2>' . t('Field mappings') . '</h2>';
+    foreach ($entity_types as $entity_type) {
+      $bundles = \Drupal::service('entity_type.bundle.info')->getBundleInfo($entity_type);
+      foreach ($bundles as $name => $attr) {
+        $rdf_mappings = rdf_get_mapping($entity_type, $name);
+        $rdf_types = $rdf_mappings->getPreparedBundleMapping();
+        if (array_key_exists('types', $rdf_types) && count($rdf_types['types']) > 0) {
+          $rdf_types = implode(', ', $rdf_types['types']);
+          $markup .= '<h3>' . $attr['label'] . ' (' . $entity_type . ')' . ', mapped to RDF type ' . $rdf_types . '</h3>';
+        }
+        else {
+          $markup .= '<h3>' . $attr['label'] . ' (' . $entity_type . ') - no RDF type mapping</h3>';
+        }
+        $fields = \Drupal::entityManager()->getFieldDefinitions($entity_type, $name);
```

Comment:
entityManager isn't a think in d9, i believe it'd be entity_field.manager. also it should probably use dependency injection since its in a controller

Reply:
OK, thanks, I'll investigate.